### PR TITLE
Enable incognito custom tab

### LINKED ISSUE #132: Links open via custom tabs are added to History even if Bromite is set to open new links in incognito mode
**Bromite version**

Version: `69.0.3497.91`
Arch: `arm64`
Android version: `Android 8.1 (LineageOS 15.1)`

**Describe the bug**
Links opened from an app that supports custom tabs are added to our history even if the new "open links in incognito tabs always" option is enabled.

**To Reproduce**

1. Clean up all your browsing history
2. Go to Settings > Privacy
3. Enable "Open links in incognito tabs always" and "Allow custom tab intents"
4. Open a link from any app that support custom apps and then close it
5. Check your browsing history

**Expected behavior**
URLs shouldn't be added to our "history".

**Screenshots**
[ScreenRecord-2018-09-17-21-32-35-1.mp4](https://www.celsoazevedo.com/files/2018/ScreenRecord-2018-09-17-21-32-35-1.mp4)

### PR DESCRIPTION
**Pull request by uazo**
this fixes #132 and https://github.com/bromite/bromite/pull/529#issuecomment-617032317 caused by https://github.com/bromite/bromite/pull/529#issuecomment-617022094

### CODE REVIEW

**Comment by uazo** [comment]
and sorry for my very bad english!

File: build/patches/Enable-incognito-custom-tab.patch [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
++++ b/chrome/android/java/src/org/chromium/chrome/browser/customtabs/CustomTabIncognitoManager.java
+@@ -50,9 +50,10 @@ public class CustomTabIncognitoManager implements NativeInitObserver, Destroyabl
+
+     @Override
+     public void onFinishNativeInitialization() {
+-        assert ChromeFeatureList.isInitialized();
++        //assert ChromeFeatureList.isInitialized();
+         if (mIntentDataProvider.isIncognito()
+-                && ChromeFeatureList.isEnabled(ChromeFeatureList.CCT_INCOGNITO)) {
++                //&& ChromeFeatureList.isEnabled(ChromeFeatureList.CCT_INCOGNITO)) {
```
Comment (csagan5) [inline build/patches/Enable-incognito-custom-tab.patch:12]: This is incorrect, the feature can be enabled by default in `chrome/browser/flags/android/chrome_feature_list.cc` and leave this untouched.

**Review by csagan5 — CHANGES_REQUESTED** [review]
How does this correlate with #529? Do they both need to be merged?

File: build/patches/Enable-incognito-custom-tab.patch (hunk 1 above)
Comment (uazo) [reply build/patches/Enable-incognito-custom-tab.patch:12]: ok, I understand

**Comment by uazo** [comment]
> How does this correlate with #529? Do they both need to be merged?

#529 introduce this new bug https://github.com/bromite/bromite/pull/529#issuecomment-617032317 solvable by this pull

**Comment by csagan5** [comment]
@uazo can you make the change to enable the feature in this PR?

**Review by csagan5 — APPROVED** [review]

**Comment by csagan5** [comment]
@uazo it can be merged for me, I copied the attribution from your git commits. See if it works for you and then I can merge it.

**Comment by uazo** [comment]
I'm still compiling ...
have you tried it? other thing, what parameter do you use in git diff to produce that file?

**Comment by csagan5** [comment]
> have you tried it?

I am also building the change.



> what parameter do you use in git diff to produce that file?

`git format-patch -1`, then remove the commit hash and `index` lines from the generated file.

**Comment by uazo** [comment]
for me doesn't work:

```
04-22 19:21:08.861 32032 32032 E AndroidRuntime: java.lang.AssertionError: swapTab shouldn't be called before setInitialTab
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.content.CustomTabActivityTabProvider.swapTab(CustomTabActivityTabProvider.java:111)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.content.CustomTabActivityTabController$1.onActivityTabChanged(CustomTabActivityTabController.java:114)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ActivityTabProvider$HintlessActivityTabObserver.onActivityTabChanged(ActivityTabProvider.java:49)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ActivityTabProvider.triggerActivityTabChangeEvent(ActivityTabProvider.java:232)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ActivityTabProvider.access$300(ActivityTabProvider.java:28)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ActivityTabProvider$2.didSelectTab(ActivityTabProvider.java:193)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.TabModelImpl.setIndex(TabModelImpl.java:535)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.IncognitoTabModel.setIndex(IncognitoTabModel.java:191)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.TabModelImpl.removeTabAndSelectNext(TabModelImpl.java:619)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.TabModelImpl.removeTab(TabModelImpl.java:103)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.IncognitoTabModel.removeTab(IncognitoTabModel.java:267)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tabmodel.TabModelSelectorImpl$3.onActivityAttachmentChanged(TabModelSelectorImpl.java:199)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tab.TabImpl.updateAttachment(TabImpl.java:332)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tab_activity_glue.ReparentingTask.detach(ReparentingTask.java:147)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.tab_activity_glue.ReparentingTask.begin(ReparentingTask.java:124)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.content.CustomTabActivityTabController.detachAndStartReparenting(CustomTabActivityTabController.java:173)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.content.CustomTabActivityNavigationController.openCurrentUrlInBrowser(CustomTabActivityNavigationController.java:274)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.CustomTabActivity.onMenuOrKeyboardAction(CustomTabActivity.java:249)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ChromeActivity.onOptionsItemSelected(ChromeActivity.java:1373)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.customtabs.CustomTabActivity.onOptionsItemSelected(CustomTabActivity.java:237)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ui.appmenu.AppMenuHandlerImpl.onOptionsItemSelected(AppMenuHandlerImpl.java:292)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ui.appmenu.AppMenu.onItemClick(AppMenu.java:366)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ui.appmenu.AppMenuAdapter.lambda$setupStandardMenuItemViewHolder$4$AppMenuAdapter(AppMenuAdapter.java:388)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at org.chromium.chrome.browser.ui.appmenu.AppMenuAdapter$$Lambda$4.onClick(Unknown Source:4)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.view.View.performClick(View.java:6597)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.view.View.performClickInternal(View.java:6574)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.view.View.access$3100(View.java:778)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.view.View$PerformClick.run(View.java:25889)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:873)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:193)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6718)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
04-22 19:21:08.861 32032 32032 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

but i'm on v84, i know it's wrong, but I only understood it later
I have to setup a better pc.

**Comment by csagan5** [comment]
@uazo I have built it here: https://fdroid.bromite.org/bromite/arm64_ChromeModernPublic_patched_v4.apk

**Comment by uazo** [comment]
It works!

**Comment by uazo** [comment]
```
adb shell am start -a android.intent.action.VIEW -d http://www.stackoverflow.com --es android.support.customtabs.extra.SESSION "1"
```

if you want try, "click cookie policy", and then "open in incognito mode"

**Comment by uazo** [comment]
please remove my mail from patch

**Comment by csagan5** [comment]
> please remove my mail from patch

@uazo it is visible in all your commits, see https://github.com/bromite/bromite/commit/2144f48a528ddc7f6be0800cbef124e3a40a9d2c.patch

I can change it but you should probably change your settings in GitHub.

**Comment by uazo** [comment]
> I can change it but you should probably change your settings in GitHub.

you are right, thanks

> git format-patch -1, then remove the commit hash and index lines from the generated file.

and thanks for that too!

**Comment by csagan5** [comment]
> you are right, thanks

Please note that your past git commits will not be updated with your new username/email.

I will proceed to merge the PR.

**Pull request closed without merging**